src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/TasksTest.kt

```diff
@@ -155,6 +155,63 @@ class TasksTest : IntegrationTestBase() {
         }
       }
     }
+
+    @Test
+    fun `Get all reallocatable tasks with taskType that doesn't exist returns 404`() {
+      `Given a User`(roles = listOf(UserRole.CAS1_WORKFLOW_MANAGER)) { user, jwt ->
+        webTestClient.get()
+          .uri("/tasks/reallocatable?type=RANDOMWORD")
+          .header("Authorization", "Bearer $jwt")
+          .exchange()
+          .expectStatus()
+          .isNotFound
+      }
+    }
+
+    @Test
+    fun `Get all reallocatable tasks with taskType of bookingAppeal returns 403`() {
+      `Given a User`(roles = listOf(UserRole.CAS1_WORKFLOW_MANAGER)) { user, jwt ->
+        webTestClient.get()
+          .uri("/tasks/reallocatable?type=bookingAppeal")
+          .header("Authorization", "Bearer $jwt")
+          .exchange()
+          .expectStatus()
+          .isForbidden
+      }
+    }
+
+    @Test
+    fun `Get all reallocatable tasks returns 200 with correct body when type is passed as parameter`() {
```

Comment:
Could we get a test for placement requests and placement applications too? Just for completeness.